### ISSUE TITLE
let statement in wrong position

### ISSUE THREAD

**Issue by LaoLittle** [rust-lang/rust-clippy#9878]
### Summary

```rust
let _ = if let 0 = 0 {
        1
    } else {
        0
    };
```

and the clippy suggested me to try this

```rust
warning: boolean to int conversion using if
 --> src/main.rs:2:13
  |
2 |       let _ = if let 0 = 0 {
  |  _____________^
3 | |         1
4 | |     } else {
5 | |         0
6 | |     };
  | |_____^ help: replace with from: `i32::from(let 0 = 0)`
  |
  = note: `#[warn(clippy::bool_to_int_with_if)]` on by default
  = note: `let 0 = 0 as i32` or `let 0 = 0.into()` can also be valid options
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_to_int_with_if
```

### Lint Name

bool_to_int_with_if

### Version

```text
rustc 1.65.0 (897e37553 2022-11-02)
binary: rustc
commit-hash: 897e37553bba8b42751c67658967889d11ecd120
commit-date: 2022-11-02
host: aarch64-apple-darwin
release: 1.65.0
LLVM version: 15.0.0
```

**Comment by Alexendoo** [comment]
Fixed by https://github.com/rust-lang/rust-clippy/pull/9714
